packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -118,6 +118,17 @@ const handle = {
         return true;
       });
 
+      // Replace `function (a, x = a.b?.#c) {}` to `function (a, x = (() => a.b?.#c)() ){}`
+      // so the temporary variable can be injected in correct scope
+      // This can be further optimized to avoid unecessary IIFE
+      if (scope.path.isPattern()) {
```

Comment:
Does this only apply in parameters? I'm not sure when a scope'a path would be the pattern.

Reply:
> Does this only apply in parameters?

Yes, if a Pattern is a `sopce.path`, its parent must be a `Function`: https://github.com/babel/babel/blob/3628c528676a91cd95bb00d56f799be53185e316/packages/babel-types/src/validators/isScope.js#L22

because a `Pattern` is not Scopeable.

However when reviewing what is authored by past me in #10912, I do realize that current `isScope` definition does not cover pattern in catch clause:

```js
try { } catch ({ e = foo?.bar }) {}
```

I will fix that in a separate PR.